qlib/workflow/recorder.py

```diff
@@ -46,6 +48,8 @@ def save_objects(self, data_name_list, local_path=None):
             list of (data, name) pairs
         local_path : str
             if provided, them save the file or directory to the artifact URI.
+        artifact_path=None : str
+            the relative path for the artifact to be stored in the URI.

```

Comment:
Why we have to specify so many arguments ( run_id=None, experiment_id=None, run_name=None, nested=False) for a recorder. 
1) Why can not recorder get these state from internal state?
2) Should  the record hide the conception of `run`?

Reply:
Scenariao